omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java

```diff
@@ -171,17 +176,21 @@ private SimpleObject parseDisposition(DispositionDescriptor dispositionDescripto
 		return simpleObject;
 	}
 	
-	private SimpleObject parseDiagnosis(DiagnosisMetadata diagnosisMetadata, Obs obs) {
-		Diagnosis diagnosis = diagnosisMetadata.toDiagnosis(obs);
+	private SimpleObject parseDiagnosis(Encounter encounter) {
+		
+		List<Diagnosis> diag = diagnosisServiceCompatibility.getPrimaryDiagnosis(encounter);
```

Comment:
hi @ibacher  @dkayiwa ,this whole line is returning a null, even on autowiring as seen in code above, could i be missing out some thing?